Makefile

```diff
@@ -195,6 +195,11 @@ endif
 $(warning Warning: Compiling in debug mode. Don't use the resulting binary in production)
 endif
 
+ifeq ($(USE_LTO), 1)
```

Comment:
Please add a comment on what USE_LTO means.

Also, should something similar be put into the CMakeLists.txt?  It seems like the two build tools should support similar options as much as possible

Reply:
> Also, should something similar be put into the CMakeLists.txt? It seems like the two build tools should support similar options as much as possible

Agreed. I looked into it a bit and found cmake provides a standardized way to build with LTO. Something like:

```
$ cmake .. -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=True -DCMAKE_POLICY_DEFAULT_CMP0069=NEW -DCMAKE_BUILD_TYPE=Release
```

The `-DCMAKE_POLICY_DEFAULT_CMP0069=NEW` could be elided if we bump the value in `cmake_minimum_required()` to 3.9. Also on my machine the link step currently fails. These require some more investigation, so I'll probably leave this PR's scope as adding support in Makefile only for now.

Reply:
> Please add a comment on what USE_LTO means.

Done.

Reply:
Can you please capture what you found in an issue for CMake so that we do not lose it.

Reply:
Sure, here it is: #7186.